FingerBlox/app/src/main/java/org/fingerblox/fingerblox/ImageDisplayActivity.java

```diff
@@ -156,26 +159,37 @@ protected void saveFeatures(String fileName){
     public void findMatch() {
         SharedPreferences preferences = getSharedPreferences("PREFS", 0);
         String[] fileNameList = preferences.getString("fileNameList", "").split(" ");
+        ArrayList<Pair<String, Integer>> matches = new ArrayList<>();
```

Comment:
Why not a `TreeMap<Integer, String>`? That way it would be automatically sorted and the code would be cleaner